### ISSUE TITLE
Cop Idea: Do not combine optional keyword arguments with double splat arguments

### ISSUE THREAD

**Issue by mvz** [rubocop/rubocop#12371]
## Description
Combining optional keyword arguments with double-splat is dangerous. Take the following code:

```ruby
def foo(a, b: 5, **rest)
  # ..
  bar(rest)
  # ...
end
```

The intent here is of course, that `rest` will never have an `a:` key. However, this could happen:

```ruby
foo(23, **args)
```

Inspecting the code would suggest `args` gets passed to `bar`, and `b` gets the default value `5`. This is true, except if somehow somewhere `args[:b]` got defined.

## Proposed solution

I would like to see a cop that forbids combining of optional keyword arguments with double splats.

```ruby
# bad
def foo(a, b: 5, **rest)
  # ...
end

# good
def foo(a, b = 5, **rest)
  # ...
end
```

## Related issues

See also #12352. If keyword arguments are preferred over regular optional arguments, the special case of additional double splat arguments should be taken into account.

**Comment by dvandersluis** [comment]
The problem I see is that there's no way of knowing that `foo(23, **args)` isn't **intentionally** setting setting a value for `b:`. 

The behaviour is similar to this example without using kwargs:

```ruby
def foo(a, b = 5)
end

foo(*args)
```

In both cases, it's unknown whether `args` overrides the default value of `b` without inspecting it, and if it does we would say that's intentional.

**Comment by mvz** [comment]
@dvandersluis that is a good point. I don't really remember the specifics of the original case well, so I'm going to close this for now.
